camera_system/game_area_info.py

```diff
@@ -0,0 +1,302 @@
+"""ゲームエリア情報モジュール.
+
+ゲームエリア情報を保持するクラスを定義している
+@author: kodama0720
+"""
+
+from typing import List, Tuple, Dict
+from robot import Robot, Direction
+from color_changer import Color
+from node import Node, NodeType
+from coordinate import Coordinate
+
+
+class GameAreaInfo:
+    """ゲームエリア情報を保持するクラス.
+
+    Attributes:
+        __cross ノードが交点
+        __middle ノードが中点
+        __block_storage ノードがブロック置き場
+        block_id_list (List[int]): ブロック(色ID)
+        base_id_list (List[int]): ベースブロック(色ID)
+        end_id (List[int]): ボーナスブロック(色ID)
+        node_list (List[Node]): ノードリスト(ノード)
+        base_color_dict (Dict{int, str}): ベースエリアの色ID(赤、黄、緑、青)
+        __east_cand_list = (List[Node]): 東の候補ノードになりうる座標リスト
+        __south_cand_list = (List[Node]): 南の候補ノードになりうる座標リスト
+        __west_cand_list = (List[Node]): 西の候補ノードになりうる座標リスト
+        __north_cand_list = (List[Node]): 北の候補ノードになりうる座標リスト
+    """
+
+    __cross = NodeType.CROSS.value
+    __middle = NodeType.MIDDLE.value
+    __block_storage = NodeType.BLOCK.value
+    block_id_list = []
+    base_id_list = []
+    end_id = []
+    node_list = [
+        Node(-1, Coordinate(0, 0), __cross), Node(-1, Coordinate(1, 0), __middle),
+        Node(-1, Coordinate(2, 0), __cross), Node(-1, Coordinate(3, 0), __middle),
+        Node(-1, Coordinate(4, 0), __cross), Node(-1, Coordinate(5, 0), __middle),
+        Node(-1, Coordinate(6, 0), __cross),
+        Node(-1, Coordinate(0, 1), __cross), Node(0, Coordinate(1, 1), __block_storage),
+        Node(-1, Coordinate(2, 1), __cross), Node(1, Coordinate(3, 1), __block_storage),
+        Node(-1, Coordinate(4, 1), __cross), Node(2, Coordinate(5, 1), __block_storage),
+        Node(-1, Coordinate(6, 1), __cross),
```

Comment:
```suggestion
        Node(-1, Coordinate(0, 1), __middle), Node(0, Coordinate(1, 1), __block_storage),
        Node(-1, Coordinate(2, 1), __middle), Node(1, Coordinate(3, 1), __block_storage),
        Node(-1, Coordinate(4, 1), __middle), Node(2, Coordinate(5, 1), __block_storage),
        Node(-1, Coordinate(6, 1), __middle),
```

かな？
入力ミスも考えると、Node側で座標からノードの種類を求める方がいいと思います。